app/controllers/sessions_controller.rb

```diff
@@ -2,54 +2,84 @@ class SessionsController < ApplicationController
   def new
   end
 
+  def shop_owner_login
```

Comment:
The session controller handles authentication for two models - users and shop owners. The `new` method is to render the users login form while the `shop_owner_login` is to render the shop owner login form. There are two different pages.